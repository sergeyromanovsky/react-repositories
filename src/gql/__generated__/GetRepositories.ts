/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL query operation: GetRepositories
// ====================================================

export interface GetRepositories_search_edges_node_App {
  __typename: "App" | "Issue" | "MarketplaceListing" | "Organization" | "PullRequest" | "User";
}

export interface GetRepositories_search_edges_node_Repository_stargazers {
  __typename: "StargazerConnection";
  /**
   * Identifies the total count of items in the connection.
   */
  totalCount: number;
}

export interface GetRepositories_search_edges_node_Repository_forks {
  __typename: "RepositoryConnection";
  /**
   * Identifies the total count of items in the connection.
   */
  totalCount: number;
}

export interface GetRepositories_search_edges_node_Repository {
  __typename: "Repository";
  /**
   * The name of the repository.
   */
  name: string;
  id: string;
  /**
   * A list of users who have starred this starrable.
   */
  stargazers: GetRepositories_search_edges_node_Repository_stargazers;
  /**
   * A list of direct forked repositories.
   */
  forks: GetRepositories_search_edges_node_Repository_forks;
  /**
   * The repository's URL.
   */
  homepageUrl: any | null;
}

export type GetRepositories_search_edges_node = GetRepositories_search_edges_node_App | GetRepositories_search_edges_node_Repository;

export interface GetRepositories_search_edges {
  __typename: "SearchResultItemEdge";
  /**
   * The item at the end of the edge.
   */
  node: GetRepositories_search_edges_node | null;
  /**
   * A cursor for use in pagination.
   */
  cursor: string;
}

export interface GetRepositories_search {
  __typename: "SearchResultItemConnection";
  /**
   * The number of repositories that matched the search query.
   */
  repositoryCount: number;
  /**
   * A list of edges.
   */
  edges: (GetRepositories_search_edges | null)[] | null;
}

export interface GetRepositories {
  /**
   * Perform a search across resources.
   */
  search: GetRepositories_search;
}

export interface GetRepositoriesVariables {
  cursor?: string | null;
}
