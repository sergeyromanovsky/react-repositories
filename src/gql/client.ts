import { ApolloClient, InMemoryCache } from '@apollo/client';

export const client = new ApolloClient({
  uri: 'https://api.github.com/graphql',
  headers: {
    Authorization: `token ${process.env.REACT_APP_GITHUB_API_TOKEN}`,
  },
  cache: new InMemoryCache()
});
