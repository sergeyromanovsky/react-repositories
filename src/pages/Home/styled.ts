import styled from 'styled-components';

export const Wrapper = styled.div`
  padding: 10rem 14rem;
  height: 100vh;
  overflow: hidden;
  box-sizing: border-box;
  display: flex;
  flex-direction: column;
`;
export const Heading = styled.h1`
  font-size: 8rem;
  line-height: 10rem;
  margin-bottom: 4rem;
`;

export const TableHeader = styled.div`
  font-size: 8rem;
  line-height: 10rem;
`;
