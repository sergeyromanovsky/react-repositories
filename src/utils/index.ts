export const getRem = (px: number) => `${px / 8}rem`;
