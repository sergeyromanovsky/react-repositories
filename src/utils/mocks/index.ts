export const ReactRepositoriesMock = {
  search: {
    repositoryCount: 129,
    edges: [
      {
        node: {
          name: 'react',
          id: 'MDEwOlJlcG9zaXRvcnkxMDI3MDI1MA==',
          stargazers: { totalCount: 151376, __typename: 'StargazerConnection' },
          forks: { totalCount: 28367, __typename: 'RepositoryConnection' },
          homepageUrl: 'https://reactjs.org',
          __typename: 'Repository',
        },
        cursor: 'Y3Vyc29yOjE=',
        __typename: 'SearchResultItemEdge',
      },
      {
        node: {
          name: 'ReactiveCocoa',
          id: 'MDEwOlJlcG9zaXRvcnkzNjA2NjI0',
          stargazers: { totalCount: 19835, __typename: 'StargazerConnection' },
          forks: { totalCount: 3408, __typename: 'RepositoryConnection' },
          homepageUrl: '',
          __typename: 'Repository',
        },
        cursor: 'Y3Vyc29yOjI=',
        __typename: 'SearchResultItemEdge',
      },
      {
        node: {
          name: 'react-native',
          id: 'MDEwOlJlcG9zaXRvcnkyOTAyODc3NQ==',
          stargazers: { totalCount: 88378, __typename: 'StargazerConnection' },
          forks: { totalCount: 18817, __typename: 'RepositoryConnection' },
          homepageUrl: 'https://reactnative.dev',
          __typename: 'Repository',
        },
        cursor: 'Y3Vyc29yOjM=',
        __typename: 'SearchResultItemEdge',
      },
      {
        node: {
          name: 'next.js',
          id: 'MDEwOlJlcG9zaXRvcnk3MDEwNzc4Ng==',
          stargazers: { totalCount: 50223, __typename: 'StargazerConnection' },
          forks: { totalCount: 7560, __typename: 'RepositoryConnection' },
          homepageUrl: 'https://nextjs.org',
          __typename: 'Repository',
        },
        cursor: 'Y3Vyc29yOjQ=',
        __typename: 'SearchResultItemEdge',
      },
      {
        node: {
          name: 'react-router',
          id: 'MDEwOlJlcG9zaXRvcnkxOTg3MjQ1Ng==',
          stargazers: { totalCount: 41006, __typename: 'StargazerConnection' },
          forks: { totalCount: 8017, __typename: 'RepositoryConnection' },
          homepageUrl: 'https://reacttraining.com/react-router/',
          __typename: 'Repository',
        },
        cursor: 'Y3Vyc29yOjU=',
        __typename: 'SearchResultItemEdge',
      },
      {
        node: {
          name: 'create-react-app',
          id: 'MDEwOlJlcG9zaXRvcnk2MzUzNzI0OQ==',
          stargazers: { totalCount: 79900, __typename: 'StargazerConnection' },
          forks: { totalCount: 18511, __typename: 'RepositoryConnection' },
          homepageUrl: 'https://create-react-app.dev',
          __typename: 'Repository',
        },
        cursor: 'Y3Vyc29yOjY=',
        __typename: 'SearchResultItemEdge',
      },
      {
        node: {
          name: 'awesome-react',
          id: 'MDEwOlJlcG9zaXRvcnkyMjY3MDg1Nw==',
          stargazers: { totalCount: 37532, __typename: 'StargazerConnection' },
          forks: { totalCount: 4198, __typename: 'RepositoryConnection' },
          homepageUrl: '',
          __typename: 'Repository',
        },
        cursor: 'Y3Vyc29yOjc=',
        __typename: 'SearchResultItemEdge',
      },
      {
        node: {
          name: 'react-boilerplate',
          id: 'MDEwOlJlcG9zaXRvcnkzMDk2OTE4OA==',
          stargazers: { totalCount: 25789, __typename: 'StargazerConnection' },
          forks: { totalCount: 5038, __typename: 'RepositoryConnection' },
          homepageUrl: 'https://www.reactboilerplate.com',
          __typename: 'Repository',
        },
        cursor: 'Y3Vyc29yOjg=',
        __typename: 'SearchResultItemEdge',
      },
      {
        node: {
          name: 'react-demos',
          id: 'MDEwOlJlcG9zaXRvcnkzMjcyNjM3Nw==',
          stargazers: { totalCount: 15167, __typename: 'StargazerConnection' },
          forks: { totalCount: 6234, __typename: 'RepositoryConnection' },
          homepageUrl: null,
          __typename: 'Repository',
        },
        cursor: 'Y3Vyc29yOjk=',
        __typename: 'SearchResultItemEdge',
      },
      {
        node: {
          name: 'react-redux-starter-kit',
          id: 'MDEwOlJlcG9zaXRvcnkzODAwMzkwMw==',
          stargazers: { totalCount: 10405, __typename: 'StargazerConnection' },
          forks: { totalCount: 2217, __typename: 'RepositoryConnection' },
          homepageUrl: '',
          __typename: 'Repository',
        },
        cursor: 'Y3Vyc29yOjEw',
        __typename: 'SearchResultItemEdge',
      },
      {
        node: {
          name: 'react-navigation',
          id: 'MDEwOlJlcG9zaXRvcnk4MDE0OTI2Mg==',
          stargazers: { totalCount: 18358, __typename: 'StargazerConnection' },
          forks: { totalCount: 3690, __typename: 'RepositoryConnection' },
          homepageUrl: 'https://reactnavigation.org',
          __typename: 'Repository',
        },
        cursor: 'Y3Vyc29yOjEx',
        __typename: 'SearchResultItemEdge',
      },
      {
        node: {
          name: 'react-select',
          id: 'MDEwOlJlcG9zaXRvcnkyMzMzODcxNg==',
          stargazers: { totalCount: 20375, __typename: 'StargazerConnection' },
          forks: { totalCount: 3303, __typename: 'RepositoryConnection' },
          homepageUrl: 'https://react-select.com/',
          __typename: 'Repository',
        },
        cursor: 'Y3Vyc29yOjEy',
        __typename: 'SearchResultItemEdge',
      },
      {
        node: {
          name: 'reaction',
          id: 'MDEwOlJlcG9zaXRvcnkxMjkyNjk2Mw==',
          stargazers: { totalCount: 10295, __typename: 'StargazerConnection' },
          forks: { totalCount: 1830, __typename: 'RepositoryConnection' },
          homepageUrl: 'https://reactioncommerce.com/',
          __typename: 'Repository',
        },
        cursor: 'Y3Vyc29yOjEz',
        __typename: 'SearchResultItemEdge',
      },
      {
        node: {
          name: 'react-use',
          id: 'MDEwOlJlcG9zaXRvcnkxNTQ5NTA5MjU=',
          stargazers: { totalCount: 13637, __typename: 'StargazerConnection' },
          forks: { totalCount: 880, __typename: 'RepositoryConnection' },
          homepageUrl: 'http://streamich.github.io/react-use',
          __typename: 'Repository',
        },
        cursor: 'Y3Vyc29yOjE0',
        __typename: 'SearchResultItemEdge',
      },
      {
        node: {
          name: 'react-redux',
          id: 'MDEwOlJlcG9zaXRvcnkzODkzNDQ0OQ==',
          stargazers: { totalCount: 19664, __typename: 'StargazerConnection' },
          forks: { totalCount: 2781, __typename: 'RepositoryConnection' },
          homepageUrl: 'https://react-redux.js.org',
          __typename: 'Repository',
        },
        cursor: 'Y3Vyc29yOjE1',
        __typename: 'SearchResultItemEdge',
      },
      {
        node: {
          name: 'storybook',
          id: 'MDEwOlJlcG9zaXRvcnk1NDE3MzU5Mw==',
          stargazers: { totalCount: 49708, __typename: 'StargazerConnection' },
          forks: { totalCount: 4717, __typename: 'RepositoryConnection' },
          homepageUrl: 'https://storybook.js.org',
          __typename: 'Repository',
        },
        cursor: 'Y3Vyc29yOjE2',
        __typename: 'SearchResultItemEdge',
      },
      {
        node: {
          name: 'react-bootstrap',
          id: 'MDEwOlJlcG9zaXRvcnkxNTQ4MDA2Nw==',
          stargazers: { totalCount: 17884, __typename: 'StargazerConnection' },
          forks: { totalCount: 2778, __typename: 'RepositoryConnection' },
          homepageUrl: 'https://react-bootstrap.github.io/',
          __typename: 'Repository',
        },
        cursor: 'Y3Vyc29yOjE3',
        __typename: 'SearchResultItemEdge',
      },
      {
        node: {
          name: 'react-admin',
          id: 'MDEwOlJlcG9zaXRvcnk2MzIyNjU4OA==',
          stargazers: { totalCount: 13246, __typename: 'StargazerConnection' },
          forks: { totalCount: 2830, __typename: 'RepositoryConnection' },
          homepageUrl: 'http://marmelab.com/react-admin',
          __typename: 'Repository',
        },
        cursor: 'Y3Vyc29yOjE4',
        __typename: 'SearchResultItemEdge',
      },
      {
        node: {
          name: 'react-virtualized',
          id: 'MDEwOlJlcG9zaXRvcnk0NTQzMTg5OA==',
          stargazers: { totalCount: 19409, __typename: 'StargazerConnection' },
          forks: { totalCount: 2321, __typename: 'RepositoryConnection' },
          homepageUrl: 'http://www.reactvirtualized.com',
          __typename: 'Repository',
        },
        cursor: 'Y3Vyc29yOjE5',
        __typename: 'SearchResultItemEdge',
      },
      {
        node: {
          name: 'react-native-web',
          id: 'MDEwOlJlcG9zaXRvcnkzNzE1MzMzNw==',
          stargazers: { totalCount: 16867, __typename: 'StargazerConnection' },
          forks: { totalCount: 1347, __typename: 'RepositoryConnection' },
          homepageUrl: 'https://necolas.github.io/react-native-web/docs',
          __typename: 'Repository',
        },
        cursor: 'Y3Vyc29yOjIw',
        __typename: 'SearchResultItemEdge',
      },
      {
        node: {
          name: 'react-starter-kit',
          id: 'MDEwOlJlcG9zaXRvcnkxODg0MDAwMw==',
          stargazers: { totalCount: 20180, __typename: 'StargazerConnection' },
          forks: { totalCount: 3827, __typename: 'RepositoryConnection' },
          homepageUrl: 'https://reactstarter.com',
          __typename: 'Repository',
        },
        cursor: 'Y3Vyc29yOjIx',
        __typename: 'SearchResultItemEdge',
      },
      {
        node: {
          name: 'awesome-react-native',
          id: 'MDEwOlJlcG9zaXRvcnkzMjk0ODg2Mw==',
          stargazers: { totalCount: 27489, __typename: 'StargazerConnection' },
          forks: { totalCount: 3251, __typename: 'RepositoryConnection' },
          homepageUrl: 'http://www.awesome-react-native.com',
          __typename: 'Repository',
        },
        cursor: 'Y3Vyc29yOjIy',
        __typename: 'SearchResultItemEdge',
      },
      {
        node: {
          name: 'react-native-elements',
          id: 'MDEwOlJlcG9zaXRvcnk2NzcwOTgwOA==',
          stargazers: { totalCount: 18946, __typename: 'StargazerConnection' },
          forks: { totalCount: 3747, __typename: 'RepositoryConnection' },
          homepageUrl:
            'https://react-native-elements.github.io/react-native-elements/',
          __typename: 'Repository',
        },
        cursor: 'Y3Vyc29yOjIz',
        __typename: 'SearchResultItemEdge',
      },
      {
        node: {
          name: 'react-native-guide',
          id: 'MDEwOlJlcG9zaXRvcnkzMzcwNjU5OA==',
          stargazers: { totalCount: 15977, __typename: 'StargazerConnection' },
          forks: { totalCount: 3491, __typename: 'RepositoryConnection' },
          homepageUrl: '',
          __typename: 'Repository',
        },
        cursor: 'Y3Vyc29yOjI0',
        __typename: 'SearchResultItemEdge',
      },
      {
        node: {
          name: 'enzyme',
          id: 'MDEwOlJlcG9zaXRvcnk0NTk0MjI3NA==',
          stargazers: { totalCount: 18862, __typename: 'StargazerConnection' },
          forks: { totalCount: 1964, __typename: 'RepositoryConnection' },
          homepageUrl: 'https://enzymejs.github.io/enzyme/',
          __typename: 'Repository',
        },
        cursor: 'Y3Vyc29yOjI1',
        __typename: 'SearchResultItemEdge',
      },
    ],
    __typename: 'SearchResultItemConnection',
  },
};
