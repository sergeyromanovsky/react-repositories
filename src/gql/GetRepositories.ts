import { gql } from '@apollo/client';

export const GET_REPOSITORIES = gql`
  query GetRepositories($cursor: String) {
    search(
      query: "react stars:>10000"
      type: REPOSITORY
      first: 25
      after: $cursor
    ) {
      repositoryCount
      edges {
        node {
          ... on Repository {
            name
            id
            stargazers {
              totalCount
            }
            forks {
              totalCount
            }
            homepageUrl
          }
        }
        cursor
      }
    }
  }
`;
