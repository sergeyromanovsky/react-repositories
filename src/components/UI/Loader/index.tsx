import React from 'react';
import { LoaderWrapper, PreLoader } from './styled';

export const LOADER_TEST_ID = 'loader';

interface IProps {
  height?: string;
}
function Loader({ height }: IProps) {
  return (
    <LoaderWrapper data-testid={LOADER_TEST_ID} height={height}>
      <PreLoader />
    </LoaderWrapper>
  );
}

export default Loader;
