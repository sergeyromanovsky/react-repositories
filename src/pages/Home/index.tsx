import React, { useCallback } from 'react';
import Loader from 'components/UI/Loader/';
import Table from 'components/Table';
import { useQuery } from '@apollo/client';
import {
  GetRepositories,
  GetRepositoriesVariables,
} from 'gql/__generated__/GetRepositories';
import { GET_REPOSITORIES } from 'gql/GetRepositories';
import { Heading, Wrapper } from './styled';

function Home() {
  const { data, fetchMore } = useQuery<
    GetRepositories,
    GetRepositoriesVariables
  >(GET_REPOSITORIES, {
    fetchPolicy: 'cache-and-network',
  });

  const loadMore = useCallback(() => {
    const list = data!.search.edges;

    return fetchMore({
      variables: {
        cursor: list![list!.length - 1]?.cursor,
      },
      updateQuery: (prev, { fetchMoreResult }) => {
        if (!fetchMoreResult) {
          return prev;
        }
        return {
          ...prev,
          search: {
            ...fetchMoreResult.search,
            edges: [...prev.search.edges!, ...fetchMoreResult.search.edges!],
          },
        };
      },
    });
  }, [data, fetchMore]);

  if (!data) {
    return <Loader height='100vh' />;
  }

  const total = data.search.repositoryCount;
  const list = data.search.edges!;

  return (
    <Wrapper>
      <Heading>Test Task</Heading>
      <Table
        list={list}
        hasNextPage={list.length < total}
        loadMoreItems={loadMore}
      />
    </Wrapper>
  );
}

export default Home;
