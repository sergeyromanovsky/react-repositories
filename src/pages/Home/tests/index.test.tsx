import React from 'react';
import { MockedProvider } from '@apollo/client/testing';
import { render } from '@testing-library/react';
import { wait, waitForDomChange } from '@testing-library/dom';
import Home from '..';
import { LOADER_TEST_ID } from 'components/UI/Loader';
import { GET_REPOSITORIES } from 'gql/GetRepositories';
import { ReactRepositoriesMock } from 'utils/mocks';
import { act } from 'react-dom/test-utils';

describe('Home page', () => {
  beforeAll(() => {
    Object.defineProperty(HTMLElement.prototype, 'offsetHeight', {
      configurable: true,
      value: 1000,
    });
    Object.defineProperty(HTMLElement.prototype, 'offsetWidth', {
      configurable: true,
      value: 1000,
    });
  });
  it('Should show loader if data is not provided', async () => {
    const { getByTestId } = render(
      <MockedProvider>
        <Home />
      </MockedProvider>
    );

    expect(getByTestId(LOADER_TEST_ID)).toBeInTheDocument();
    await wait();
  });

  it('Should fetch data and match snapshot', async () => {
    const mocks = [
      {
        request: {
          query: GET_REPOSITORIES,
        },
        result: {
          data: ReactRepositoriesMock,
        },
      },
      {
        request: {
          query: GET_REPOSITORIES,
          variables: { cursor: 'Y3Vyc29yOjI1' },
        },
        result: {
          data: ReactRepositoriesMock,
        },
      },
    ];

    const { getByText, baseElement } = render(
      <MockedProvider mocks={mocks}>
        <Home />
      </MockedProvider>
    );

    await act(async () => {
      await waitForDomChange();
      expect(getByText('react')).toBeInTheDocument();
      expect(baseElement).toMatchSnapshot();
    });
  });
});
