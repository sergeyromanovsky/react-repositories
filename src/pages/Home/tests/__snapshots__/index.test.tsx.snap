// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page Should fetch data and match snapshot 1`] = `
<body>
  <div>
    <div
      class="sc-gKsewC iHuIWH"
    >
      <h1
        class="sc-iBPRYJ cAPLjL"
      >
        Test Task
      </h1>
      <div
        class="sc-dlfnbm cYiCfa"
      >
        <div
          class="sc-hKgILt cnvXHn"
        >
          name
        </div>
        <div
          class="sc-hKgILt cnvXHn"
        >
          üåü stars
        </div>
        <div
          class="sc-hKgILt cnvXHn"
        >
          üç¥ forks
        </div>
      </div>
      <div
        style="height: 100%; width: 100%;"
      >
        <div
          style="overflow: visible; height: 0px; width: 0px;"
        >
          <div
            style="position: relative; height: 1000px; width: 1000px; overflow: auto; will-change: transform; direction: ltr;"
          >
            <div
              style="height: 1508px; width: 100%;"
            >
              <div
                class="sc-jSgupP dViByq"
                style="position: absolute; left: 0px; top: 0px; height: 58px; width: 100%;"
              >
                <div
                  class="sc-eCssSg kNKupr"
                >
                  <a
                    href="https://reactjs.org"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    react
                  </a>
                </div>
                <div
                  class="sc-eCssSg kNKupr"
                >
                  151,376
                </div>
                <div
                  class="sc-eCssSg kNKupr"
                >
                  28,367
                </div>
              </div>
              <div
                class="sc-jSgupP hQqFdP"
                style="position: absolute; left: 0px; top: 58px; height: 58px; width: 100%;"
              >
                <div
                  class="sc-eCssSg kNKupr"
                >
                  <a
                    href=""
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    ReactiveCocoa
                  </a>
                </div>
                <div
                  class="sc-eCssSg kNKupr"
                >
                  19,835
                </div>
                <div
                  class="sc-eCssSg kNKupr"
                >
                  3,408
                </div>
              </div>
              <div
                class="sc-jSgupP ikAyZY"
                style="position: absolute; left: 0px; top: 116px; height: 58px; width: 100%;"
              >
                <div
                  class="sc-eCssSg kNKupr"
                >
                  <a
                    href="https://reactnative.dev"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    react-native
                  </a>
                </div>
                <div
                  class="sc-eCssSg kNKupr"
                >
                  88,378
                </div>
                <div
                  class="sc-eCssSg kNKupr"
                >
                  18,817
                </div>
              </div>
              <div
                class="sc-jSgupP fonxks"
                style="position: absolute; left: 0px; top: 174px; height: 58px; width: 100%;"
              >
                <div
                  class="sc-eCssSg kNKupr"
                >
                  <a
                    href="https://nextjs.org"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    next.js
                  </a>
                </div>
                <div
                  class="sc-eCssSg kNKupr"
                >
                  50,223
                </div>
                <div
                  class="sc-eCssSg kNKupr"
                >
                  7,560
                </div>
              </div>
              <div
                class="sc-jSgupP kdGhst"
                style="position: absolute; left: 0px; top: 232px; height: 58px; width: 100%;"
              >
                <div
                  class="sc-eCssSg kNKupr"
                >
                  <a
                    href="https://reacttraining.com/react-router/"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    react-router
                  </a>
                </div>
                <div
                  class="sc-eCssSg kNKupr"
                >
                  41,006
                </div>
                <div
                  class="sc-eCssSg kNKupr"
                >
                  8,017
                </div>
              </div>
              <div
                class="sc-jSgupP cJglLV"
                style="position: absolute; left: 0px; top: 290px; height: 58px; width: 100%;"
              >
                <div
                  class="sc-eCssSg kNKupr"
                >
                  <a
                    href="https://create-react-app.dev"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    create-react-app
                  </a>
                </div>
                <div
                  class="sc-eCssSg kNKupr"
                >
                  79,900
                </div>
                <div
                  class="sc-eCssSg kNKupr"
                >
                  18,511
                </div>
              </div>
              <div
                class="sc-jSgupP jZToaN"
                style="position: absolute; left: 0px; top: 348px; height: 58px; width: 100%;"
              >
                <div
                  class="sc-eCssSg kNKupr"
                >
                  <a
                    href=""
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    awesome-react
                  </a>
                </div>
                <div
                  class="sc-eCssSg kNKupr"
                >
                  37,532
                </div>
                <div
                  class="sc-eCssSg kNKupr"
                >
                  4,198
                </div>
              </div>
              <div
                class="sc-jSgupP lcHHNU"
                style="position: absolute; left: 0px; top: 406px; height: 58px; width: 100%;"
              >
                <div
                  class="sc-eCssSg kNKupr"
                >
                  <a
                    href="https://www.reactboilerplate.com"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    react-boilerplate
                  </a>
                </div>
                <div
                  class="sc-eCssSg kNKupr"
                >
                  25,789
                </div>
                <div
                  class="sc-eCssSg kNKupr"
                >
                  5,038
                </div>
              </div>
              <div
                class="sc-jSgupP wMvqE"
                style="position: absolute; left: 0px; top: 464px; height: 58px; width: 100%;"
              >
                <div
                  class="sc-eCssSg kNKupr"
                >
                  <a
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    react-demos
                  </a>
                </div>
                <div
                  class="sc-eCssSg kNKupr"
                >
                  15,167
                </div>
                <div
                  class="sc-eCssSg kNKupr"
                >
                  6,234
                </div>
              </div>
              <div
                class="sc-jSgupP dPSHRb"
                style="position: absolute; left: 0px; top: 522px; height: 58px; width: 100%;"
              >
                <div
                  class="sc-eCssSg kNKupr"
                >
                  <a
                    href=""
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    react-redux-starter-kit
                  </a>
                </div>
                <div
                  class="sc-eCssSg kNKupr"
                >
                  10,405
                </div>
                <div
                  class="sc-eCssSg kNKupr"
                >
                  2,217
                </div>
              </div>
              <div
                class="sc-jSgupP eaWqSv"
                style="position: absolute; left: 0px; top: 580px; height: 58px; width: 100%;"
              >
                <div
                  class="sc-eCssSg kNKupr"
                >
                  <a
                    href="https://reactnavigation.org"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    react-navigation
                  </a>
                </div>
                <div
                  class="sc-eCssSg kNKupr"
                >
                  18,358
                </div>
                <div
                  class="sc-eCssSg kNKupr"
                >
                  3,690
                </div>
              </div>
              <div
                class="sc-jSgupP eHPjav"
                style="position: absolute; left: 0px; top: 638px; height: 58px; width: 100%;"
              >
                <div
                  class="sc-eCssSg kNKupr"
                >
                  <a
                    href="https://react-select.com/"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    react-select
                  </a>
                </div>
                <div
                  class="sc-eCssSg kNKupr"
                >
                  20,375
                </div>
                <div
                  class="sc-eCssSg kNKupr"
                >
                  3,303
                </div>
              </div>
              <div
                class="sc-jSgupP cdyLAj"
                style="position: absolute; left: 0px; top: 696px; height: 58px; width: 100%;"
              >
                <div
                  class="sc-eCssSg kNKupr"
                >
                  <a
                    href="https://reactioncommerce.com/"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    reaction
                  </a>
                </div>
                <div
                  class="sc-eCssSg kNKupr"
                >
                  10,295
                </div>
                <div
                  class="sc-eCssSg kNKupr"
                >
                  1,830
                </div>
              </div>
              <div
                class="sc-jSgupP kWzpWE"
                style="position: absolute; left: 0px; top: 754px; height: 58px; width: 100%;"
              >
                <div
                  class="sc-eCssSg kNKupr"
                >
                  <a
                    href="http://streamich.github.io/react-use"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    react-use
                  </a>
                </div>
                <div
                  class="sc-eCssSg kNKupr"
                >
                  13,637
                </div>
                <div
                  class="sc-eCssSg kNKupr"
                >
                  880
                </div>
              </div>
              <div
                class="sc-jSgupP jQqFAR"
                style="position: absolute; left: 0px; top: 812px; height: 58px; width: 100%;"
              >
                <div
                  class="sc-eCssSg kNKupr"
                >
                  <a
                    href="https://react-redux.js.org"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    react-redux
                  </a>
                </div>
                <div
                  class="sc-eCssSg kNKupr"
                >
                  19,664
                </div>
                <div
                  class="sc-eCssSg kNKupr"
                >
                  2,781
                </div>
              </div>
              <div
                class="sc-jSgupP cfMaOh"
                style="position: absolute; left: 0px; top: 870px; height: 58px; width: 100%;"
              >
                <div
                  class="sc-eCssSg kNKupr"
                >
                  <a
                    href="https://storybook.js.org"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    storybook
                  </a>
                </div>
                <div
                  class="sc-eCssSg kNKupr"
                >
                  49,708
                </div>
                <div
                  class="sc-eCssSg kNKupr"
                >
                  4,717
                </div>
              </div>
              <div
                class="sc-jSgupP jEAQcd"
                style="position: absolute; left: 0px; top: 928px; height: 58px; width: 100%;"
              >
                <div
                  class="sc-eCssSg kNKupr"
                >
                  <a
                    href="https://react-bootstrap.github.io/"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    react-bootstrap
                  </a>
                </div>
                <div
                  class="sc-eCssSg kNKupr"
                >
                  17,884
                </div>
                <div
                  class="sc-eCssSg kNKupr"
                >
                  2,778
                </div>
              </div>
              <div
                class="sc-jSgupP euFvMh"
                style="position: absolute; left: 0px; top: 986px; height: 58px; width: 100%;"
              >
                <div
                  class="sc-eCssSg kNKupr"
                >
                  <a
                    href="http://marmelab.com/react-admin"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    react-admin
                  </a>
                </div>
                <div
                  class="sc-eCssSg kNKupr"
                >
                  13,246
                </div>
                <div
                  class="sc-eCssSg kNKupr"
                >
                  2,830
                </div>
              </div>
              <div
                class="sc-jSgupP ewmUxT"
                style="position: absolute; left: 0px; top: 1044px; height: 58px; width: 100%;"
              >
                <div
                  class="sc-eCssSg kNKupr"
                >
                  <a
                    href="http://www.reactvirtualized.com"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    react-virtualized
                  </a>
                </div>
                <div
                  class="sc-eCssSg kNKupr"
                >
                  19,409
                </div>
                <div
                  class="sc-eCssSg kNKupr"
                >
                  2,321
                </div>
              </div>
              <div
                class="sc-jSgupP fRtwUc"
                style="position: absolute; left: 0px; top: 1102px; height: 58px; width: 100%;"
              >
                <div
                  class="sc-eCssSg kNKupr"
                >
                  <a
                    href="https://necolas.github.io/react-native-web/docs"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    react-native-web
                  </a>
                </div>
                <div
                  class="sc-eCssSg kNKupr"
                >
                  16,867
                </div>
                <div
                  class="sc-eCssSg kNKupr"
                >
                  1,347
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="resize-triggers"
        >
          <div
            class="expand-trigger"
          >
            <div
              style="width: 1001px; height: 1001px;"
            />
          </div>
          <div
            class="contract-trigger"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;
